### PR TITLE
Bug 1823707 - Refactor settingsAdvancedItemsTest UI tests

### PR DESCRIPTION
**Pull request by AndiAJ**
Bug 1823707 - Refactor settingsAdvancedItemsTest UI tests
`settingsAdvancedItemsTest` successfully passed 100x on Firebase. ✅ 

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1823707

### CODE REVIEW

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsRobot.kt [hunk 1]
```diff
@@ -159,7 +190,79 @@ class SettingsRobot {
     fun verifyAboutFirefoxPreview() = assertTrue(aboutFirefoxHeading().waitForExists(waitingTime))
     fun verifyGooglePlayRedirect() = assertGooglePlayRedirect()
 
-    fun verifySettingsOptionSummary(summary: String) = itemContainingText(summary)
+    fun verifySettingsOptionSummary(setting: String, summary: String) {
```
[the remark this answers is not in the document]
Comment (AndiAJ) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsRobot.kt:193]: You can actually reuse it anytime.
You'll check the summary update based on the corresponding setting option. 

Lets use the Homepage option as an example.
Check the default summary:
`verifySettingsOptionSummary("Homepage", "Open on homepage after four hours")`

Change the setting to "Homepage" and verify the summary:
`verifySettingsOptionSummary("Homepage", "Open on homepage")`
            
Change the setting to "Last tab" and verify the summary:
`verifySettingsOptionSummary("Homepage", "Open on last tab")`

Change the setting to "Homepage after four hours of inactivity" and verify the summary:
`verifySettingsOptionSummary("Homepage", "Open on homepage after four hours")`

[the remark this answers is not in the document]
Comment (AndiAJ) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsRobot.kt:193]: Talked to @sv-ohorvath , decided to keep things simple ☺️

[the remark this answers is not in the document]
Comment (sv-ohorvath) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsRobot.kt:193]: looks good

**Review by sv-ohorvath — APPROVED** [review]

**Review by AaronMT — APPROVED** [review]
LGTM

**Pull request merged**